major: various fixes, codebase improvements/cleanup, packages/sdk update

bryanpth opened the pull request:
### What does this PR do?

#### Updates

- `.NET SDK` from `7.0` to `8.0`
- `C# language` from `9.0` to `12.0`
- and all packages to their latest available versions. (e.g: discord.net, etc..)

#### Fixes

- Fixed `/compare` not properly returning when the same player was asked to be compared.
- Fixed a typo in the alliance rename command output.
- Fixed non-awaited warning for compensation tasks.
- Fixed pagniation not working in DMs.

#### Improvements

- Improved many publicly available commands response ouput.
- Improved almost every moderation commands and logging response ouput.
- Improved various commands response strings.
- Added usernames for authorized discord user ids, this allows to easily identify which user has empowered permissions.
- Added missing `readonly` modifiers on various files.
- Made slight improvements to code readability on various files.
- Made others various miscellaneous changes that improves codebase quality.

#### Documentation

- Updates `README.md` for a clearer output.

#### Codebase cleanup

- Deleted unused Dashboard project.
- Added `.editorconfig` to ensure proper code quality.
- Added `.gitattributes` to ensure normalized line endings to lf internally.

#### Continuous integration

- Deleted previous `console.yml` and `dashboard.yml` workflows.
- Added `ci.yaml`, which ensure proper code formatting and warning/error free application compiling.

--

- [x] Tested all publicly available commands.

bryanpth committed:
chore: add usernames for authorized discord user ids

This change makes it easier to identify which user has private empowered permissions when looking at the files.

bryanpth committed:
chore: add missing `readonly` modifiers for immutable fields

bryanpth committed:
chore: remove unused exception variable

bryanpth committed:
refactor: make methods static where applicable

bryanpth committed:
refactor: use null-coalescing assignment where applicable

bryanpth committed:
refactor: use standard C# constructor for new string creation

bryanpth committed:
refactor: use single quote when searching for a single character

- As per [CA1847](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1847).

bryanpth committed:
fix: typo `contributers` to `contributors`

bryanpth committed:
refactor: update `/help` command embed

bryanpth committed:
refactor: simplify object instantiation with target-typed `new()`

bryanpth committed:
refactor: use pattern matching

bryanpth committed:
chore: update `/help` command description

bryanpth committed:
refactor: update `/serverstats` command embed

bryanpth committed:
refactor: update ChannelInfoModule.cs

- Updated embed
- Renamed lambda parameters for clarity

bryanpth committed:
refactor: update PaginatorServices.cs

- Updated `HandleUpdateMessagePagesAsync` embed
- Added embed timestamp restoring
- Renamed lambda parameters for clarity

bryanpth committed:
refactor: update LogService.cs

- Updated `GetEmbedForLog` returned embeds.
- Renamed `victim` to `target`.
- Formatting and identation changes for code readability.
- Converted `GetColorBasedOnAction` switch statement to a switch expression for code readability.

bryanpth committed:
refactor: update DevModule.cs

- Updated `ShowAllStatsAsync` returned embed
- Updated `AddModerationModuleToGuildAsync` returned embed
- Renamed lambda parameters for clarity

bryanpth committed:
refactor: update TelemetryModule.cs

- Updated embeds builders to use `withTitle()`
- Minor changes to indentation for code readability

bryanpth committed:
refactor: update StagingModule.cs

- Updated `/overview` returned response embed.
- Updated `/reloadrules` command returned response embed.
- Updated `/reset` command returned response embed.
- Renamed lambda parameters for clarity.

bryanpth committed:
chore: rename various lambda parameters for clarity

bryanpth committed:
refactor: update GLModule.cs

- Updated `/status` command output embed.
- Updated `/compare` command output embed.
- Various lambda parameters renamed for clarity.
- Code formatting and indentation changes for readability.

bryanpth committed:
refactor: update GLService.cs

- Updated various strings.
- Code formatting and indentation changes for readability.

bryanpth committed:
refactor: update moderation service/module

- Updated all moderation commands output embeds.
- Updated many commands description.
- Renamed various lambda parameters for clarity.
- Code formatting and indentation changes for readability.
- More miscellaneous changes.

bryanpth committed:
docs: update README.md

svr333 commented on src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs:
```diff
@@ -8,7 +8,15 @@ namespace AdvancedBot.Core.Commands.Preconditions
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
```suggestion
```

svr333 replied:
Feel free to add yourself though

bryanpth replied:
> Feel free to add yourself though

thanks, will do!

svr333 commented on src/AdvancedBot.Core/Commands/Preconditions/RequireStagingList.cs:
```diff
@@ -8,7 +8,16 @@ namespace AdvancedBot.Core.Commands.Preconditions
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
```suggestion
```

svr333 commented on src/AdvancedBot.Core/Commands/TopModule.cs:
```diff
@@ -19,23 +19,27 @@ public class TopModule : InteractionModuleBase<SocketInteractionContext>
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
```suggestion
```

svr333 commented on src/AdvancedBot.Core/Services/LogService.cs:
```diff
@@ -65,115 +65,116 @@ public Embed GetEmbedForLog(Log log, User victim)
-                default:
-                    return Color.Blue;
-            }
+                LogAction.Ban or LogAction.Reset or LogAction.KickOffline or LogAction.RemoveBeta or 
+                LogAction.RemoveUserFromAlliance or LogAction.EnableMaintenance or LogAction.ReloadRules
+                    => Color.Red,
+
+                LogAction.Unban or LogAction.AddBeta or LogAction.GiveRole
+                    => Color.Green,
+
+                LogAction.UpdateEmail or LogAction.UpdateName or LogAction.GetFull or LogAction.GetChipsBought or
+                LogAction.MakeUserAllianceOwner or LogAction.GetWarlogs
+                    => new Color(15710778),
+
+                LogAction.AddChips or LogAction.AddItem or LogAction.RenameAlliance
+                    => Color.Blue,
+
+                _ => Color.Blue
+            };
```
Damn this is new syntax for me, looks cool

bryanpth committed:
chore: remove kibble user id from authorized users lists

Co-authored-by: svr333 <vanrompaey.senne@protonmail.com>

bryanpth committed:
fix: suppress non-awaited warnings

Discarding the result of compensation tasks to suppress non-awaited warnings, assuming these tasks aren't awaited on purpose

bryanpth committed:
refactor: update profile command embed

bryanpth committed:
refactor update stats command embed

- refactor `FormatNumber` to use switch expression instead of regular if else.
- fixed typo in number formatting, there was 2 checks for 10k<, leading to the last one never being reached
- Various visual improvements to the returned embed

bryanpth committed:
add missing `readonly` modifiers

bryanpth committed:
feat: replace `setGameAsync` with `setActivctyAsync` for the bot rich presence

bryanpth committed:
refactor: use switch expression instead of if else statement

- matching earlier changes to `GLService.cs`

bryanpth committed:
docs: add line break in README.md

bryanpth committed:
style: remove unnecessary blank line

bryanpth committed:
chore: add my user id in authorized users ids

svr333 commented on src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs:
```diff
@@ -442,110 +485,156 @@ public async Task EnableMaintenanceAsync(uint minutes = 60)
         public async Task CompensateChips(uint amount)
         {
-            GLClient.Production.CompensateChips(amount);
+            _ = GLClient.Production.CompensateChips(amount);
```
```suggestion
            await GLClient.Production.CompensateChips(amount);
```

svr333 commented on src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs:
```diff
@@ -442,110 +485,156 @@ public async Task EnableMaintenanceAsync(uint minutes = 60)
         public async Task CompensateItems(string sku, uint amount)
         {
-            GLClient.Production.CompensateItems(sku, amount);
+            _ = GLClient.Production.CompensateItems(sku, amount);
```
```suggestion
            await GLClient.Production.CompensateItems(sku, amount);
```

svr333 reviewed (commented):
Im pretty sure I just forgot to await these, rest of new stuff looks good

svr333 committed:
Update src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs

svr333 committed:
Update src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs

bryanpth commented:
> Im pretty sure I just forgot to await these, rest of new stuff looks good

oh, i assumed compensation task would take a while so you weren't awaiting them to not block the rest of the execution, thanks for checking it though

svr333 commented:
Oh actually now that you mention it that is probably what I did, whoops, will revert those commits

bryanpth committed:
revert: "Update src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs"

This reverts commit e4ed959fb8cb8266463789aaa8a62add8115b5fc.

bryanpth committed:
revert: "Update src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs"

This reverts commit 21568f06a19056fa3b0b972cfb4be79be1f42937.

bryanpth commented:
> Oh actually now that you mention it that is probably what I did, whoops, will revert those commits

alright, reverted those for ya dw

bryanpth committed:
fix: remove unused directive

bryanpth committed:
chore: merge branch 'master' into 'refactor/cleanup'

bryanpth committed:
bump: update .NET, C# and packages versions

- Updated .NET SDK from 7.0 to 8.0.
- Updated C# language version from 9.0 to 11.0.
- Updated Discord.Net from 3.10.0 to 3.15.3.
- Updated Microsoft.Extensions.DependencyInjection from 6.0.0 to 8.0.0.
- Updated LiteDB from 5.0.13 to 5.0.21.
- Updated Microsoft.Bcl.AsyncInterfaces from 6.0.0 to 8.0.0.

bryanpth committed:
ci: update .NET version to 8 on workflows

bryanpth committed:
refactor: update obsolete attribute

bryanpth committed:
fix: command stopped working in DM

Previously, some commands embeds were using the guild icon as thumbnail url, which lead to command stopping to work in DM since no guild icon was available.
These embeds will now use the bot avatar within DM to resolve that issue.

tldr: skill issue resolved

bryanpth committed:
refactor: update string comparison to use .Equals

- Slightly more efficient
- Cleaner syntax

bryanpth committed:
feat: use GetDisplayAvatarUrl() for avatar retrieving

Previously, we were using GetAvatarUrl(), which was returning the user's custom avatar, but if the user had not set any custom avatar, it would return "null", which could potentially lead to invalid url when used.

This replaces it with GetDisplayAvatarUrl(), which returns the custom avatar if set, and the default avatar if not set.

bryanpth committed:
fix: properly returning if same player were compared

- currently, even if you ask to compare the same player, it still gives you the comparison, as well as the "error" message.

this adds a return statement to prevent this

bryanpth commented:
`AdvancedBot.Dashboard` doesn't seem to be used as it seems to be startup template right now.
Do you mind if I was to delete it to fruther clean up the codebase?

svr333 commented:
Originally, 3 years ago the dashboard was being worked on by twobits to have an admin panel for GLR on it.
But yeah I dont think we actually need one for now so you can delete it

bryanpth committed:
ci: remove dashboard workflow

bryanpth committed:
cleanup: delete `AdvancedBot.Dashboard` project

bryanpth committed:
refactor: use top-level statement

bryanpth committed:
refactor: update "serverstats" command embed response

bryanpth committed:
fix: pagination not working in DM

Previously, using a command that was returning a paginated response would also return an error "Object reference not set to an instance of an object.".

This commit fixes that issue, which allows pagination to be used on DM interactions.

Also added thumbnail restoration, and cleaned up the syntax a little bit.

bryanpth committed:
chore: update leaderboards commands responses

bryanpth committed:
fix: typo in leaderboard embed footer

bryanpth committed:
chore: add thumbnail & footer to moderation leaderboard cmd embed

bryanpth committed:
chore: run codecleanup

bryanpth committed:
chore: remove obsolete methods

bryanpth committed:
chore: add .editorconfig

bryanpth committed:
style: format codebase to improve code quality

bryanpth committed:
ci: refactor continuous integration

bryanpth committed:
fix(ci): prevent duplicated workflow on pull requests

bryanpth committed:
chore: add .gitattributes file

bryanpth committed:
bump: update to C# 12

bryanpth committed:
bump: update Discord.NET to 3.16.0

bryanpth committed:
refactor: simplify lists/collections initialization

bryanpth committed:
fix(embed): update emote as someone renamed them on gl server

bryanpth committed:
feat: use collection expression for embeds

bryanpth committed:
chore: update reset failed response message

bryanpth committed:
chore: update profiles embed titles for consistency

bryanpth committed:
chore: simplify ServiceCollectionExt.cs

bryanpth committed:
chore: format changes on PrefixCheckerExt.cs

bryanpth committed:
chore: prefer file scoped namespace

bryanpth committed:
chore: ensure codebase match updated preferences

bryanpth committed:
chore: ensure codebase match updated preferences

Forgot to push that formatted file earlier

svr333 reviewed (approved):
Everything looks good to me, thanks for all the refactoring!
